src/extensions/common/useBufferedDispatch.ts

```diff
@@ -0,0 +1,82 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from "react"
```

Comment:
useBufferedDispatch is a new custom hook that delays the dispatch of actions. This provides a much more responsive experience for interacting with a widget, because the actions take immediate effect on the active component, but the updates are propagated less often to the global application. For example, timeseries scrolling.